airflow/providers/elasticsearch/CHANGELOG.rst

```diff
@@ -27,6 +27,17 @@
 Changelog
 ---------
 
+5.1.0
+.....
+
+Breaking changes
+~~~~~~~~~~~~~~~~
+
+.. note::
+  Upgrade to ElasaticSearch 8. The ElasticsearchTaskHandler & ElasticsearchSQLHook will now use ElasticSearch 8 package.
+  This also means that the release drops support for ElasticSearch 7 and below and will no longer work with
+  ElasticSearch database that is below version 8.
```

Comment:
I am not sure if this PR is a breaking change?
Airflow wise what exactly gets broken here?
Updating upstream package version by itself is not a breaking change.

Reply:
Sorry - what I'm trying to say is that this release will drop support for ElasticSearch database version 7 or below since `elasticsearch 8` will not work with DB version below 8 (i.e. airflow users should also upgrade their ES database to version 8 to ensure compatibility). I wrote this because I read the following doc and learn that `Elasticsearch language clients are only backwards compatible with default distributions and without guarantees made`

https://elasticsearch-py.readthedocs.io/en/stable/

Airflow-wise, nothing should really get broken

Reply:
Yeah. @eladkal is right - dependency change is not a reason for major bump - even if it is such a "big" dependency. This very clear in  https://semver.org/#what-should-i-do-if-i-update-my-own-dependencies-without-changing-the-public-api:

> What should I do if I update my own dependencies without changing the public API?

> That would be considered compatible since it does not affect the public API. Software that explicitly depends on the same dependencies as your package should have their own dependency specifications and the author will notice any conflicts. Determining whether the change is a patch level or minor level modification depends on whether you updated your dependencies in order to fix a bug or introduce new functionality. We would usually expect additional code for the latter instance, in which case it’s obviously a minor level increment.